compiler/damlc/tests/daml-test-files/ContractKeyNotEffective.daml

```diff
@@ -1,7 +1,9 @@
--- @ERROR range=14:1-14:19; contract 0053db8d12a2aebf02e9d3c2dc653c7b8f61914eeef91b2d6e08b5b066925592a4 not effective, but we found its key!
+-- @ERROR range=16:1-16:19; Couldn't see contract with key 'alice'
```

Comment:
I'm not sure this one is still testing the same thing, wdyt @remyhaemmerle-da?

Reply:
Right, if you wanted to specifically test for that error, it comes down to the same issue of `queryContractKey` not returning errors

Reply:
Good catch @akrmn.
Here we really want to test the `fetchByKey` primitive.
We need to rewrite that with an helper template and a `createAndExercise` command

Reply:
Ah right okay, will do

Reply:
A note about this test - the goal was to be able to take this suite and run it on other ledgers to ensure they act the same - however this test uses passTime to move backwards in time, which is only support on the IDE ledger, not other ledgers, according to the [docs](https://docs.daml.com/daml-script/api/Daml-Script.html#function-daml-script-settime-25783)

As such, we'd need one of the following:
- Enforce some way to setTime on any ledger that uses this suite, when in some sort of test mode
- Remove this test
- Add a flag to this test that guards it to only be run when we are able to setTime

Reply:
Good point. However a the goal of this PR is to turn scenario to script in the context of compiler integration test, we will consider the wider goal in a second step. 
The PR is huge enough to ignore  `passTime` for now